File: split/newrule.pas
function newrule: halfword;
var
  p: halfword;
begin
  p := getnode(4);
  mem[p].hh.b0 := 2;
  mem[p].hh.b1 := 0;
  mem[p + 1].int := -1073741824;
  mem[p + 2].int := -1073741824;
  mem[p + 3].int := -1073741824;
  newrule := p;
end;

File: cpp/newrule.cpp
#include "newrule.h"
#include "getnode.h"

halfword newrule(void)
{
	auto p = getnode(rule_node_size);
	type(p) = rule_node;
	subtype(p) = 0;
	width(p) = null_flag;
	depth(p) = null_flag;
	height(p) = null_flag;
	return p;
}

